### ISSUE TITLE
Request menu with available identities

### ISSUE THREAD

**Issue by ciuly** [hultmann/multifox#161]
Currently, having multiple identities and wanting to go to a specific one, requires me to click on "new" until I get to it. I have about 10 identities, so this is rather annoying.
The alternative is to always keep all profiles visible in the list, not just the last 2-3 most used.
I'm fine with either option.

And of course, the "new" button should actually be a new profile, not "next" profile.

**Comment by hultmann** [comment]
I believe it's fixed in Multifox 3.0 beta. Please reopen this if the bug is still around.

https://github.com/hultmann/multifox/issues/150#issuecomment-53949852

**Comment by hultmann** [comment]
from https://github.com/hultmann/multifox/issues/150#issuecomment-55936090:

> not working right. It has the exact same menu entries as release version, but the menu window is as tall as my screen and is all empty. I restarted FF, same thing.

hmmm... Multifox menu should list all profiles. A profile exists while there are cookies associated to it. So it seems cookies are disappearing somehow. What extensions do you use?

**Comment by ciuly** [comment]
The cookies are not gone. If I click on the "new" button, it goes to the next profile just fine. It just doesn't list them all.

the extensions are:

Adblock Plus    2.6.4   true    {d10d0bf8-f5b5-c8b4-a8b2-2b9879e08c5d}
DownThemAll!    2.0.17  true    {DDC359D1-844A-42a7-9AA1-88A850A938A8}
FEBE    8.0.5   true    {4BBDD651-70CF-4821-84F8-2B918CF89CA3}
Firebug 2.0.4   true    firebug@software.joehewitt.com
FireStorage Plus!   1.7 true    firestorageplus@nickbelhomme.com
Greasemonkey    2.2 true    {e4a8a97b-f2ed-450b-b12d-ee082ba24781}
Multifox    3.0.0b1 true    multifox@hultmann
Phrase Highlighter  1.9.1   true    phrasehighlighter@techtoast.co.uk
Saved Password Editor   2.7.3   true    savedpasswordeditor@daniel.dawson
Search Highlight    1.0.1   true    searchhighlight@piwisoft.com
SearchWP    2.7 true    {3e270ac3-8936-43fb-ad20-b4685172a83d}
Session Manager 0.8.1.5 true    {1280606b-2510-4fe0-97ef-9b5a22eafe30}
Speed Dial  0.9.6.16    true    {64161300-e22b-11db-8314-0800200c9a66}
Tab Mix Plus    0.4.1.4 true    {dc572301-7619-498c-a57d-39143191b318}
TooManyTabs 1.4.5   true    TooManyTabs@visibotech.com
YouTube Video and Audio Downloader  0.3.9.1 true    feca4b87-3be4-43da-a1b1-137c24220968@jetpack
Microsoft .NET Framework Assistant  0.0.0   false   {20a82645-c095-46ed-80e3-08825760534b}
PrivDog 2.2.0.14    false   PrivDog@AdTrustMedia.com

**Comment by hultmann** [comment]
When do you click "New", the "new" profile (i.e. a profile that should be listed) is still logged in? (e.g. to Google/Github etc)

**Comment by ciuly** [comment]
yes. Let's call that "next" profile :), so clicking "New" button, the first time after installing beta, opened the "next" profile. All logged in and working good. At that point I had 3 profiles "activated" (the first 2 I use on a daily basis, the 3d on every other week). So those are the 3 I use the most. I have another 6 or so profiles which I use very rarely, and those were not logged into gmail as I tried them now. So it's not clear if that was because I haven't logged in them for 1 month or so, or because the cookies vanished. 

Now, once I walked through them all and got to profile "10", that's a new one, all profiles are listed. The menu window is still as tall as the screen with a lot of empty space in it. Having closed the tab in which I opened the profiles one by one with the "new" button, I opened a new tab, while in the "default" profile, and clicking "new" now opened profile "11". At this point I deleted 10 and 11.

So there is the issue of the menu window being too tall.

There is also the issue of installing this beta on a system with multiple profiles in which some profiles were not used for a long time and hence are not displayed in the list. These will not be displayed until you go through them all. Because I haven't used them in a very long time, it is not clear if (some) cookies have disappeared or not.
If I login to gmail on one of these rarely used profiles, close the tab and now select it from the list (as new button creates new profiles now). the account remains logged in as it should.

So I am thinking that the actual issue is with displaying all profiles, no matter when they were last used. It's not clear to me at this point if it's a reminiscent from the previous build and is actually solved now, or the profiles will go missing again once they gather enough unused days. If that happens, I'll post here.

So if you fix the menu height issue (so I don't open another ticket for it, unless you want me to), you can close this ticket for now.

Thanks.

**Comment by hultmann** [comment]
Popup height is fixed in 3.0b2: https://addons.mozilla.org/en-US/firefox/addon/multifox/ [Development Channel]

for beta 3, I'm going to change the approach to list profiles, there are other bugs about this.

**Comment by ciuly** [comment]
confirmed the height fix. Thanks.

**Comment by hultmann** [comment]
It is fixed in Multifox 3.0.0b4 (to be released) by https://github.com/hultmann/multifox/commit/d010e3ec58f78a2c95cdddb180c9bd2be0313d99

Now all profiles are saved in a preference, so they never disappear after you create them.

**Comment by ciuly** [comment]
Ok, was preparing to update this ticket as today I noticed out of the 9 profiles, I can only see 6 of them.

It'll take another month to confirm the fix though. After I install the release, of course. Will reopen if not fixed.

Thanks.
